Upgrade to OTel v0.116.0 (#2314)
grafana/alloy: docs/sources/reference/components/otelcol/otelcol.exporter.loadbalancing.md

The loadbalancer configures the exporter for the signal types supported by the `routing_key`.

The `timeout` argument is similar to the top-level `queue` and `retry` [blocks][] for `otelcol.exporter.loadbalancing` itself.
It helps to re-route data into a new set of healthy backends.
This is especially useful for highly elastic environments like Kubernetes, 
where the list of resolved endpoints changes frequently due to deployments and scaling events. 

> 
> > **EXPERIMENTAL**: Metrics support in `otelcol.exporter.loadbalancing` is an [experimental][] feature.
> > Experimental features are subject to frequent breaking changes, and may be removed with no equivalent replacement.
> > The `stability.level` flag must be set to `experimental` to use the feature.

Reviewer: ```suggestion

[blocks]: #blocks

> **EXPERIMENTAL**: Metrics support in `otelcol.exporter.loadbalancing` is an [experimental][] feature.
> Experimental features are subject to frequent breaking changes, and may be removed with no equivalent replacement.
> The `stability.level` flag must be set to `experimental` to use the feature.

[experimental]: https://grafana.com/docs/release-life-cycle/
```

Missing the blocks link definition, and when we included the custom experimental text we forgot to add the link definition for the release life cycle.

protocol > otlp > client > keepalive | [keepalive][]     | Configures keepalive settings for the gRPC client.                                | no
protocol > otlp > queue              | [queue][]         | Configures batching of data before sending.                                       | no
protocol > otlp > retry              | [retry][]         | Configures retry mechanism for failed requests.                                   | no
queue                                | [queue][]         | Configures batching of data before sending to the `otlp > protocol` exporter.     | no
retry                                | [retry][]         | Configures retry mechanism for failed requests to the `otlp > protocol` exporter. | no
debug_metrics                        | [debug_metrics][] | Configures the metrics that this component generates to monitor its state.        | no

The `>` symbol indicates deeper levels of nesting. For example, `resolver > static`
refers to a `static` block defined inside a `resolver` block.

There are two types of [queue][] and [retry][] blocks:

> * The ones under `protocol > otlp`. This is useful for temporary problems with a specific backend, like transient network issues.
> * The ones top-level ones for `otelcol.exporter.loadbalancing` itself. 

Reviewer: ```suggestion
* The queue and retry blocks under `protocol > otlp`. This is useful for temporary problems with a specific backend, like transient network issues.
* The top-level queue and retry blocks for `otelcol.exporter.loadbalancing`. 
```

  Those configuration options provide capability to re-route data into a new set of healthy backends.

>   This are useful for highly elastic environments like Kubernetes, 

Reviewer: ```suggestion
  This is useful for highly elastic environments like Kubernetes, 
```
